boltdb/version.go

```diff
@@ -0,0 +1,81 @@
+package boltdb
+
+import (
+	"bytes"
+	"context"
+
+	bolt "go.etcd.io/bbolt"
+)
+
+func updateVersion(
+	ctx context.Context,
+	tx *bolt.Tx,
+	h string,
+	r, c, n []byte,
+) (bool, error) {
+	// Retrieve/create the handler bucket.
+	b, err := bucket(tx, TopBucket, h)
+	if err != nil {
+		return false, err
+	}
+
+	// If the "current" version is not nil (i.e. the first call for the handler)
+	// and different to the value in the resource bucket, that means the current
+	// version was not correct.
+	if c1 := b.Get(r); c1 != nil && !bytes.Equal(c, c1) {
```

Comment:
I don't **think** this should check for `nil`, just equality.  As per the Dogma spec, `nil` and `[]byte{}` must be treated as equivalent.

```suggestion
	if !bytes.Equal(b.Get(r), c) {
```